recipe/build.sh

```diff
@@ -5,21 +5,24 @@
 if [[ "$cuda_compiler_version" == "None" ]]; then
   export FORCE_CUDA=0
 else
-  export TORCH_CUDA_ARCH_LIST="3.5;5.0+PTX"
   if [[ ${cuda_compiler_version} == 9.0* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;7.0"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;7.0+PTX"
   elif [[ ${cuda_compiler_version} == 9.2* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
   elif [[ ${cuda_compiler_version} == 10.* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
   elif [[ ${cuda_compiler_version} == 11.0* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
   elif [[ ${cuda_compiler_version} == 11.1* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
   elif [[ ${cuda_compiler_version} == 11.2* ]]; then
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+  elif [[ ${cuda_compiler_version} == 11.8* ]]; then
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+  elif [[ ${cuda_compiler_version} == 12.* ]]; then
+      export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
   else
-      export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
```

Comment:
Instead of having a default, would recommend turning this into an error. That way if a yet-to-be-considered CUDA version shows up, we can reassess what needs to be done then (if anything). Also the error will point to what the next steps would be

This lines up with [what PyTorch does]( https://github.com/conda-forge/pytorch-cpu-feedstock/blob/7c7a57b7515eaeda67d3879b56b68466f38f0b0d/recipe/build_pytorch.sh#L154-L157 ) as well as other PyTorch dependent packages

```suggestion
        echo "unsupported cuda version. edit build.sh"
        exit 1
```

Reply:
I agree, since this is in the build script I rather get a loud "something as wrong" than some fallback behavior since we ought to fix whatever issue crops up, this isn't a satellite that has to work in orbit no matter what.